### ISSUE TITLE
Order Creation: Accessibility → Back button in Variations list has no label

### ISSUE THREAD

**Issue by shiki** [woocommerce/woocommerce-ios#6072]
When using VoiceOver, this button just says “Button”. I would expect something like “Back Button”. 

<img src="https://user-images.githubusercontent.com/198826/152435073-4e0b6e05-0fa1-426f-af5d-d1297d8c17c6.png" width="300">
